python/composio_swe/benchmark/visualize.py

```diff
@@ -0,0 +1,189 @@
+import argparse
+import glob
+import json
+
+
+def extract_details(agent_logs):
+    data = []
+    for log in agent_logs:
+        from pprint import pprint
+
+        pprint(log)
+        if log["agent_action"] in ["agent_finish", "final_patch"]:
+            agent_action = "agent_finish"
+            agent_output = log["agent_output"]
+        else:
+            agent_action = json.loads(log["agent_action"])
+            tool_name = agent_action.get("tool", "N/A")
+            tool_input = agent_action.get("tool_input", "N/A")
+            tool_output = log.get("tool_output", "N/A")
+            agent_thought = agent_action.get("log", "No thoughts recorded")
+
+        data.append(
+            {
+                "tool_name": tool_name or agent_action,
+                "tool_input": tool_input,
+                "tool_output": tool_output or agent_output,
+                "agent_thought": agent_thought,
+            }
+        )
+    return data
+
+
+def generate_html_1(issues_data, output_file):
+    html_content = """
+    <html>
+    <head>
+        <title>Agent Action Report by Issue</title>
+        <style>
+            body { font-family: Arial, sans-serif; margin: 40px; }
+            table { width: 100%; border-collapse: collapse; }
+            th, td { border: 1px solid #dddddd; padding: 8px; text-align: left; }
+            th { background-color: #f2f2f2; }
+        </style>
+    </head>
+    <body>
+        <h1>Agent Action Report by Issue</h1>
+    """
+    for issue, data in issues_data.items():
+        html_content += f"<h2>{issue}</h2>"
+        html_content += """
+        <table>
+            <tr>
+                <th>Tool Name</th>
+                <th>Tool Input</th>
+                <th>Tool Output</th>
+                <th>Agent Thought</th>
+            </tr>
+        """
+        for entry in data:
+            html_content += f"""
+                <tr>
+                    <td>{entry['tool_name']}</td>
+                    <td>{entry['tool_input']}</td>
+                    <td>{entry['tool_output']}</td>
+                    <td>{entry['agent_thought']}</td>
+                </tr>
+            """
+        html_content += "</table>"
+    html_content += """
+    </body>
+    </html>
+    """
+
+    with open(output_file, "w") as file:
+        file.write(html_content)
+    print(f"HTML report generated: {output_file}")
+
+
+def format_thought(thought):
+    # print(thought)
+    # Initial setup for default values if any of the parts are missing
+    action = "No action specified"
```

Comment:
The `format_thought` function could be improved by using regular expressions to extract the different parts of the thought. This would make the parsing more robust and less prone to errors if the format of the thought changes slightly. Consider using the `re` module for this purpose.